### PR TITLE
expanded codesnip section

### PR DESCRIPTION
**Pull request by nwatx**

### CODE REVIEW

File: content/1_General/Working_MDX.mdx [hunk 1]
```diff
@@ -420,16 +430,40 @@ can be hovered over.
...
 Code blocks are separated by three backticks, just like in normal markdown.
-Additionally, we have support for collapsible code snippets:
+Additionally, we have support for collapsible code snippets. An example for how
+to use them can be found below:
+
+**With title**
+
+```
+Before
+//BeginCodeSnip{Optional Code Snippet Title}
```
Comment (maggie-j-liu) [inline content/1_General/Working_MDX.mdx:440]: i think the main point of confusion on how to use codesnips is caused by the fact that these codesnips are replaced by actual codesnips, so people don't see the syntax of putting them in a mdx file

idk if there's a way to escape them so they aren't replaced??

Comment (nwatx) [reply content/1_General/Working_MDX.mdx:440]: image

yeah it looks like this on my end

Comment (maggie-j-liu) [reply content/1_General/Working_MDX.mdx:440]: ohh nvm
i think the codesnips don't get replaced if they're not in a codeblock with language

File: content/1_General/Working_MDX.mdx [hunk 2]
```diff
@@ -420,16 +430,40 @@ can be hovered over.
...
+My snippet code goes here
```
Comment (SansPapyrus683) [inline content/1_General/Working_MDX.mdx:441]:
```suggestion
Code snippet goes here

File: content/1_General/Working_MDX.mdx [hunk 3]
```diff
@@ -420,16 +430,40 @@ can be hovered over.
...
 
-Another example of a snippet without titles:
+**Without title:**
 
+```
+//BeginCodeSnip{}
+My snippet code goes here
+//EndCodeSnip
+
+My non-snippet code goes here
```
Comment (SansPapyrus683) [inline content/1_General/Working_MDX.mdx:463]:
```suggestion
Other code goes here
```

File: content/1_General/Working_MDX.mdx [hunk 4]
```diff
@@ -420,16 +430,40 @@ can be hovered over.
...
 //EndCodeSnip
+After
+```
 
-Another example of a snippet without titles:
+**Without title:**
 
+```
+//BeginCodeSnip{}
+My snippet code goes here
```
Comment (SansPapyrus683) [inline content/1_General/Working_MDX.mdx:460]: Code snippet goes here

File: content/1_General/Working_MDX.mdx [hunk 5]
```diff
@@ -420,16 +430,40 @@ can be hovered over.
...
+You can indent the entire BeginCodeSnip block (including the BeginCodeSnip line) and it will function as expected
+//EndCodeSnip
+After
+```
 
 ```cpp
+Before
 //BeginCodeSnip{Optional Code Snippet Title}
 My snippet code goes here
```
Comment (SansPapyrus683) [inline content/1_General/Working_MDX.mdx:450]:
```suggestion
Code snippet goes here
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/1_General/Working_MDX.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply content/1_General/Working_MDX.mdx:460]:
```suggestion
Code snippet goes here
```

**Review by SansPapyrus683 — APPROVED** [review]
pog

File: content/1_General/Working_MDX.mdx [hunk 6]
```diff
@@ -320,11 +329,12 @@ author: Nathan Wang
...
-problem. Make sure to also update the `kind` of the `solutionMetadata` to `'internal'` for any
-associated problems. We assume that if there is an internal solution, we should
-use it; therefore, the build will throw an error if there is an internal
-solution but the `solutionMetadata`'s `kind` isn't set to `'internal'`. The
-[next module](/general/adding-solution) describes how to add a new solution.
+problem. Make sure to also update the `kind` of the `solutionMetadata` to
+`'internal'` for any associated problems. We assume that if there is an internal
+solution, we should use it; therefore, the build will throw an error if there is
+an internal solution but the `solutionMetadata`'s `kind` isn't set to
+`'internal'`. The [next module](/general/adding-solution) describes how to add a
```
Comment (maggie-j-liu) [inline content/1_General/Working_MDX.mdx:336]: adding solution is not the next module

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: content/1_General/Working_MDX.mdx (hunk 6 above)
Comment (bqi343) [reply content/1_General/Working_MDX.mdx:336]: @nwatx

Comment (nwatx) [inline content/1_General/Working_MDX.mdx:336]:
```suggestion
`'internal'`. The [adding solution module](/general/adding-solution) describes how to add a
```

**Review by maggie-j-liu — APPROVED** [review]

**Pull request merged**